### PR TITLE
Add VisaSponsorshipApplicationDeadlineAt & VisaSponsorshipApplicationDeadlineRequired steps

### PR DESCRIPTION
**Pull request by gms-gs**
## Context

This change completes the visa sponsorship branch in the add-course wizard so providers can explicitly set whether they need a visa-related application deadline, and if so, enter that deadline date.  
It replaces the previous direct jump to the next section with a clearer decision path and adds date validation to prevent invalid or out-of-window deadlines.

## Changes proposed in this pull request

- Adds a new `visa_sponsorship_application_deadline_required` step (yes/no) shown when visa sponsorship is required.
- Adds a new `visa_sponsorship_application_deadline_at` date step, including validation for:
  - blank/partial dates
  - non-numeric values
  - invalid calendar dates
  - out-of-range dates (must be between start-of-cycle/today and cycle apply deadline)
- Updates wizard navigation so visa sponsorship paths now route through these steps before `start_date`.
- Persists new state keys in the wizard store/repository:
  - `visa_sponsorship_application_deadline_required`
  - `visa_sponsorship_application_deadline_at`
- Adds/updates locale strings for both new pages and validation messages.
- Expands test coverage across:
  - step specs
  - system specs
  - wizard next/previous step specs
  - repository/state store specs

## Guidance to review

- Walk through add-course flows where visa sponsorship is required:
  - fee-funded route (`visa_sponsorship`)
  - salary/apprenticeship route (`skilled_worker_visa`)
- Verify routing:
  - sponsorship required -> deadline required step
  - deadline required = yes -> deadline date step
  - deadline required = no -> `start_date`
- On date step, test edge cases:
  - empty fields, partial fields, letters, invalid date (e.g. 31/02), and after apply deadline
- Confirm copy/errors come from locale entries and match expected content.
- Suggested checks:
  - `bundle exec rspec spec/system/publish/courses/add_course_wizard/visa_sponsorship_application_deadline_required_spec.rb`
  - `bundle exec rspec spec/system/publish/courses/add_course_wizard/visa_sponsorship_application_deadline_at_spec.rb`
  - `bundle exec rspec spec/wizards/add_course_wizard/wizard/next_step_spec.rb spec/wizards/add_course_wizard/wizard/previous_step_spec.rb`

## Checklist

- [ ] I have moved hard-coded strings to locale files.
- [ ] I have removed the usage of `data-qa` attributes in HTML files and updated the corresponding tests.

### CODE REVIEW

**Comment by gms-gs** [comment]
[Implement visa-sponsorship-application-deadline required and date step](https://trello.com/c/vryMczrQ/1730-implement-visa-sponsorship-application-deadline-required-and-date-step)

File: config/locales/en/publish/course_wizard.yml [hunk 1]
```diff
@@ -168,6 +168,32 @@ en:
         errors:
           can_sponsor_skilled_worker_visa:
             blank: Select if candidates can get a sponsored Skilled Worker visa
+      visa_sponsorship_application_deadline_required:
+        caption: Add course
+        heading: Is there a deadline for applications that require visa sponsorship?
+        question: Is there a deadline for applications that require visa sponsorship?
```
Comment (tomas-stefano) [inline config/locales/en/publish/course_wizard.yml]: visa_sponsorship_application_deadline_required defines both heading: and question:  with identical text, but the view  (_visa_sponsorship_application_deadline_required.html.erb) renders heading and uses a hidden legend — question is never read (it's a copy-paste from skilled_worker_visa, where question is used).

File: app/wizards/course_wizard/steps/visa_sponsorship_application_deadline_at.rb [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      end
+
+    private
+
+      def date_present
+        error_type = if [year, month, day].all?(&:blank?)
+                       :all_blank
+                     elsif [year, month, day].any?(&:blank?)
+                       :some_blank
+                     elsif [year, month, day].any? { |entry| entry.match?(/\A[a-zA-Z'-]*\z/) }
```
Comment (tomas-stefano) [inline app/wizards/course_wizard/steps/visa_sponsorship_application_deadline_at.rb]: entry.match?(/\A[a-zA-Z'-]*\z/) only flags fully alphabetic fields. A value like "1a" passes date_present, then set_date does "1a".to_i # => 1, silently coercing.  Edge case, and inherited from the original form.

File: app/wizards/course_wizard/steps/visa_sponsorship_application_deadline_at.rb [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      def formatted_first_valid_datetime
+        if Time.zone.now.after? start_of_cycle
+          "today"
+        else
+          start_of_cycle.to_fs(:govuk_date_and_time)
+        end
+      end
+
+      def start_of_cycle
+        @start_of_cycle ||= wizard.recruitment_cycle.application_start_date.end_of_day.change(hour: 9)
```
Comment (tomas-stefano) [inline app/wizards/course_wizard/steps/visa_sponsorship_application_deadline_at.rb]: application_start_date.end_of_day.change(hour: 9) — change(hour: 9) resets
  min/sec/nsec to 0, so the .end_of_day is overwritten and the result is simply
  09:00 on the start date. Harmless but confusing; drop .end_of_day when you
  consolidate.

File: app/wizards/course_wizard/steps/visa_sponsorship_application_deadline_at.rb (hunk 2 above)
Comment (gms-gs) [reply app/wizards/course_wizard/steps/visa_sponsorship_application_deadline_at.rb]: Amazing catch!

**Review by tomas-stefano — APPROVED** [review]
